packages/next/src/server/use-cache/resume-data-cache.ts

```diff
@@ -0,0 +1,223 @@
+import {
+  arrayBufferToString,
+  stringToUint8Array,
+} from '../app-render/encryption-utils'
+import type { CacheEntry } from '../lib/cache-handlers/types'
+import type { CachedFetchValue } from '../response-cache/types'
+
+type ResumeStoreSerialized = {
+  version: 1
+  store: {
+    cache: {
+      [key: string]: any
+    }
+    fetch: {
+      [key: string]: any
+    }
+  }
+}
+
+interface CacheStore<T, S = T> {
+  get(key: string): T | undefined
+  set(key: string, value: T): void
+  entries(): Promise<[string, S][]>
+  seal(): void
+}
+
+class FetchCacheStore implements CacheStore<CachedFetchValue> {
+  private readonly store: Map<string, CachedFetchValue>
+
+  /**
+   * Whether the store is immutable.
+   */
+  private immutable: boolean = false
+
+  constructor(entries?: Iterable<[string, CachedFetchValue]>) {
+    if (entries) {
+      this.immutable = true
+      this.store = new Map(entries)
+    } else {
+      this.store = new Map()
+    }
+  }
+
+  public seal() {
+    this.immutable = true
+  }
+
+  public set(key: string, value: CachedFetchValue): void {
+    if (this.immutable) return
```

Comment:
seems like this should error rather than silently not set the value. we should control out write such that we never attempt to set a cache on a sealed cache instance ideally. Is there a practical reason why we have to seal this way?